pkg/operator/handlers/dapr_handler_test.go

```diff
@@ -129,6 +131,48 @@ func TestCreateDaprServiceAppIDAndMetricsSettings(t *testing.T) {
 	assert.Equal(t, "", service.ObjectMeta.Annotations["prometheus.io/path"])
 }
 
+func TestPatchDaprService(t *testing.T) {
+	testDaprHandler := getTestDaprHandler()
+
+	s := runtime.NewScheme()
+	err := scheme.AddToScheme(s)
+	assert.NoError(t, err)
+	testDaprHandler.Scheme = s
+
+	cli := fake.NewClientBuilder().WithScheme(s).Build()
+	testDaprHandler.Client = cli
+
+	ctx := context.Background()
+	myDaprService := types.NamespacedName{
+		Namespace: "test",
+		Name:      "test",
+	}
+	deployment := getDeployment("test", "true")
+
+	err = testDaprHandler.createDaprService(ctx, myDaprService, deployment)
+	assert.NoError(t, err)
+	var actualService corev1.Service
+	err = cli.Get(ctx, myDaprService, &actualService)
+	assert.NoError(t, err)
+	assert.Equal(t, "test", actualService.ObjectMeta.Annotations[appIDAnnotationKey])
+	assert.Equal(t, "true", actualService.ObjectMeta.Annotations["prometheus.io/scrape"])
+	assert.Equal(t, "/", actualService.ObjectMeta.Annotations["prometheus.io/path"])
+	assert.Len(t, actualService.OwnerReferences, 1)
+	assert.Equal(t, "Deployment", actualService.OwnerReferences[0].Kind)
+	assert.Equal(t, "app", actualService.OwnerReferences[0].Name)
+
+	err = testDaprHandler.patchDaprService(ctx, myDaprService, deployment, actualService)
+	assert.NoError(t, err)
+	err = cli.Get(ctx, myDaprService, &actualService)
+	assert.NoError(t, err)
+	assert.Equal(t, "test", actualService.ObjectMeta.Annotations[appIDAnnotationKey])
+	assert.Equal(t, "true", actualService.ObjectMeta.Annotations["prometheus.io/scrape"])
+	assert.Equal(t, "/", actualService.ObjectMeta.Annotations["prometheus.io/path"])
+	assert.Len(t, actualService.OwnerReferences, 1)
+	assert.Equal(t, "Deployment", actualService.OwnerReferences[0].Kind)
+	assert.Equal(t, "app", actualService.OwnerReferences[0].Name)
```

Comment:
Please add test for delete case, ensure service is deleted as well.

Reply:
The service with OwnerReference field will be deleted by kubernetes garbage collector in controller manager, but there is no real k8s cluster introduced in the current unit test, we just create the service with fake client. So is there any suggestions for the deleting case?

Reply:
We have e2e tests, you can add it here.